IPython/lib/pretty.py

```diff
@@ -109,6 +109,18 @@ def _repr_pretty_(self, p, cycle):
 
 from typing import Dict
 
+# Allow pretty-printing of functions with PEP-649 annotations
+try:
+    # Actually an annotationlib symbol imported by inspect
+    # but if inspect has it that should mean it can use it
+    from inspect import Format
+except ImportError:
+    pass
+else:
+    from functools import partial
+    signature = partial(signature, annotation_format=Format.FORWARDREF)
```

Comment:
I would prefer have a `if sys.version_info>= (3,14):...`, this way once we drop 3.13 we can remove conditional.